modules/login/php/authenticatetoken.class.inc

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Implements the user password reset page for LORIS.
+ *
+ * PHP version 5
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+namespace LORIS\login;
+
+/**
+ * Implements the user password reset page for LORIS.
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Dave MacFarlane <david.macfarlane2@mcgill.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class AuthenticateToken extends \NDB_Form
+{
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @return bool whether the user has access
+     */
+    function _hasAccess()
+    {
+        return true;
+    }
+
+    /**
+     * CommonPageSetup sets up the common form elements which
+     * are on the page, regardless of HTTP request method.
+     *
+     * @return none
+     */
+    function setup()
+    {
+        $DB       =& \Database::singleton();
+        $factory  = \NDB_Factory::singleton();
+        $settings = $factory->settings();
+        $baseURL  = $settings->getBaseURL();
+
+        $token =$_GET['token'];
+
+        //validate token and load password change page
+        $validToken = $DB->pselectRow(
+            'SELECT *, TIMESTAMPDIFF(HOUR,time,CURRENT_TIMESTAMP) as token_age 
```

Comment:
@driusan should we expire the tokens ? if so, I need to start using this token_age variable